zerver/lib/actions.py

```diff
@@ -817,6 +817,44 @@ def do_send_messages(messages):
     # intermingle sending zephyr messages with other messages.
     return already_sent_ids + [message['message'].id for message in messages]
 
+def do_send_typing_notification(notification, stopped):
+    # type: (Dict[str, Any], bool) -> None
+    if stopped is True:
+        event_type = 'stopped_typing_notification'
+    else:
+        event_type = 'typing_notification'
+    event = dict(
+            type            = event_type,
+            sender          = notification['sender'],
+            recipient       = notification['recipient'])
+    user = notification['recipient']
+    users = [user]
+    send_event(event, users)
```

Comment:
Does this only support singleton recipients, or do you support group PMs as well?  This logic certainly doesn't handle group PMs properly.  It'd be great to make sure the backend handles those correctly, both through fixing this logic and adding a few tests.